code/core/src/main/java/com/adobe/marketing/mobile/services/MessagingDelegate.java

```diff
@@ -23,38 +24,80 @@ public interface MessagingDelegate {
      *
      * @param message {@link FullscreenMessage} that is being displayed
      */
+    @Deprecated
     default void onShow(final FullscreenMessage message) {
         Log.debug(ServiceConstants.LOG_TAG, "MessagingDelegate", "Fullscreen message shown.");
     }
 
+    /**
+     * Invoked when a {@code Showable} message is displayed.
+     *
+     * @param message {@link Showable} message that is being displayed
+     */
+    default void onShow(final Showable message) {
+        Log.debug(ServiceConstants.LOG_TAG, "MessagingDelegate", "Message shown.");
+    }
+
     /**
      * Invoked when a message is dismissed.
      *
      * @param message {@link FullscreenMessage} that is being dismissed
      */
+    @Deprecated
     default void onDismiss(final FullscreenMessage message) {
         Log.debug(ServiceConstants.LOG_TAG, "MessagingDelegate", "Fullscreen message dismissed.");
     }
 
+    /**
+     * Invoked when a {@code Showable} message is dismissed.
+     *
+     * @param message {@link Showable} message that is being dismissed
+     */
+    default void onDismiss(final Showable message) {
+        Log.debug(ServiceConstants.LOG_TAG, "MessagingDelegate", "Message shown.");
+    }
+
     /**
      * Used to determine if a message should be shown.
      *
      * @param message {@link FullscreenMessage} that is about to get displayed
      * @return true if the message should be displayed, false otherwise
      */
+    @Deprecated
     boolean shouldShowMessage(final FullscreenMessage message);
 
+    /**
+     * Used to determine if a message should be shown.
+     *
+     * @param message {@link Showable} message that is about to get displayed
+     * @return true if the message should be displayed, false otherwise
+     */
+    default boolean shouldShowMessage(final Showable message) {
```

Comment:
Does this apply only to messages? Given that this interface now handles everything in terms of presentables/showables - what are your thoughts on naming this to be more generic (.. `shouldShow(...)`)  ?

Reply:
i considered changing the naming of the function but the function itself is within a `MessagingDelegate` interface. I feel like `showShowMessage` fits better?

Reply:
👍🏼 Got it. I can go either way.